File: 9-palindrome-number/palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        reverse = 0
        y = x
        
        while y != 0:
            digit = y % 10
            reverse = reverse * 10 + digit
            y //= 10
        return reverse == x

        

File: 9-palindrome-number/palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        int dup = x;
        int revnum = 0;
        if (x < 0)
          return false;
        while (x != 0) 
        {  
            int ld = x % 10;
            if((revnum > INT_MAX/10 )|| (revnum < INT_MIN/10))
            {
                return false;
            }
            revnum = (revnum * 10) + ld;
            x /= 10;
             
        }
         
         if (dup == revnum) 
            return true;
         else
           return false;
         
    }
};